neo/Consensus/ConsensusService.cs

```diff
@@ -151,14 +153,33 @@ private void Log(string message, LogLevel level = LogLevel.Info)
             Plugin.Log(nameof(ConsensusService), level, message);
         }
 
+        private void SendRegenerationMessageIfNecessary()
+        {
+            // Allow the primary or any node that has received the prepare request to regenerate other nodes trying
+            // to request change view.
+            // Note: In the future, we may want to limit how many nodes will send a regeneration msg if not the primary.
+            if (context.MyIndex == context.PrimaryIndex || context.Preparations[context.PrimaryIndex] != null )
+                localNode.Tell(new LocalNode.SendDirectly { Inventory = context.MakeRegenerationMessage() });
+        }
+
         private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
         {
-            if (context.State.HasFlag(ConsensusState.CommitSent))
-                return;
             if (message.NewViewNumber <= context.ExpectedView[payload.ValidatorIndex])
                 return;
+
+            if (message.NewViewNumber <= context.ViewNumber)
```

Comment:
`ChangeView` messages may be delayed. This cannot be considered a newly joined consensus node.

Maybe we can create two new messages: `RecoveryRequest` and `RecoveryResponse`.

A node send `RecoveryRequest` in `OnStart()`. And when the other nodes receive it, they reply the `RecoveryResponse` (which is currently named `RegenerationMessage`).

Reply:
It shouldn't be a problem to send a `ChangeView` message with the same view number the node is currently on, and use it as a `RecoveryRequest`.

Reply:
If a `ChangeView` is received delayed and a node responds by sends out a regeneration message, it won't hurt anything.

Reply:
But message delays often occur. This will increase the network load.

Reply:
Maybe the `ChangeView` message could have a flag for whether it wants a `RecoveryMessage` returned if it is received on the same view that it is requesting to change to instead of adding a new message type.

Reply:
Changed to only trigger the regeneration message if the `NewViewNumber` is less instead of less than or equal to the current` ViewNumber`.